src/pre_process/m_perturbation.fpp

```diff
@@ -0,0 +1,616 @@
+    ! complex general matrix
+
+    use ieee_arithmetic
+
+    ! ==========================================================================
+    ! ==========================================================================
+
+    implicit none
+
+    integer :: mixlayer_nvar ! Number of variables in linear stability analysis solver for mixing layer
+    integer, allocatable, dimension(:) :: mixlayer_var ! Index of variables in linear stability analysis solver
+
+contains
+
+    subroutine s_initialize_perturbation_module()
+
+        if (mixlayer_perturb) then
+            if (model_eqns == 2 .and. num_fluids == 1) then
+                mixlayer_nvar = 5 ! 1 continuity + 3 momentum + 1 energy
+                allocate (mixlayer_var(mixlayer_nvar))
+
+                mixlayer_var(1) = contxb
+                mixlayer_var(2) = momxb
+                mixlayer_var(3) = momxb + 1
+                mixlayer_var(4) = momxb + 2
+                mixlayer_var(5) = momxb + 3
+            end if
+        end if
+
+    end subroutine s_initialize_perturbation_module
+
+    subroutine s_perturb_sphere(q_prim_vf)
+        type(scalar_field), dimension(sys_size), intent(inout) :: q_prim_vf
+        integer :: i, j, k, l !< generic loop operators
+
+        real(kind(0d0)) :: perturb_alpha
+        real(kind(0d0)) :: alpha_unadv
+        real(kind(0d0)) :: rand_real
+        call random_seed()
+
+        do k = 0, p
+            do j = 0, n
+                do i = 0, m
+                    call random_number(rand_real)
+
+                    perturb_alpha = q_prim_vf(E_idx + perturb_sph_fluid)%sf(i, j, k)
+
+                    ! Perturb partial density fields to match perturbed volume fraction fields
+                    !    IF ((perturb_alpha >= 25d-2) .AND. (perturb_alpha <= 75d-2)) THEN
+                    if ((perturb_alpha /= 0d0) .and. (perturb_alpha /= 1d0)) then
+
+                        ! Derive new partial densities
+                        do l = 1, num_fluids
+                            q_prim_vf(l)%sf(i, j, k) = q_prim_vf(E_idx + l)%sf(i, j, k)*fluid_rho(l)
+                        end do
+
+                    end if
+                end do
+            end do
+        end do
+
+    end subroutine s_perturb_sphere
+
+    subroutine s_perturb_surrounding_flow(q_prim_vf)
+        type(scalar_field), dimension(sys_size), intent(inout) :: q_prim_vf
+        integer :: i, j, k, l !<  generic loop iterators
+
+        real(kind(0d0)) :: perturb_alpha
+        real(kind(0d0)) :: rand_real
+        call random_seed()
+
+        ! Perturb partial density or velocity of surrounding flow by some random small amount of noise
+        do k = 0, p
+            do j = 0, n
+                do i = 0, m
+
+                    perturb_alpha = q_prim_vf(E_idx + perturb_flow_fluid)%sf(i, j, k)
+                    ! IF (perturb_alpha == 1d0) THEN
+                    ! Perturb partial density
+                    !    CALL RANDOM_NUMBER(rand_real)
```

Comment:
should you delete these lines

Reply:
Same as the comment below. 

_`s_perturb_sphere` and `s_perturb_surrounding_flow` are not written by myself. I guess they were written many years ago. I just moved these subroutines to `m_perturbation.fpp` along with instability wave subroutines because it seems more appropriate to put the "perturbation" stuff together in a separate module. I don't think any of the example cases use these subroutines. Maybe these subroutines can be removed then?_

Reply:
I just mean delete the commented out lines

Reply:
deleted!